Display unversioned requirers on Naming Policy page

irushchyshyn opened the pull request:
This PR contains UI for  #279 and #283.

Index view:
![image](https://cloud.githubusercontent.com/assets/5603118/25862502/93f4329e-34e8-11e7-85d7-36f92683cc6c.png)


Naming policy view:
![image](https://cloud.githubusercontent.com/assets/5603118/25862481/7c099ac0-34e8-11e7-90aa-e5fe33bf17e2.png)

hroncok commented:
I like how it looks. However I have a bad feeling about "unversioned requires", in my mind, that means the package requires `python2-requests` instead of `python2-requests >= 2.2`; not `python-requests` vs. `python2-requests`. Can we call it "Misnamed requires" instead?

irushchyshyn commented:
Good point @hroncok , thanks! I will change it.

hroncok commented on data/naming.yaml:
```diff
@@ -21,25 +21,30 @@
+    some build-time or run-time requirements are of type <i>python-foo</i>,
+    although the names with a versioned Python prefix
+    (e.g. <i>python2-foo</i>) are available
+  short_description: The package uses wrong naming scheme in Requires
```
(Build)Requires maybe?

irushchyshyn replied:
Amended to 14f4a06

encukou commented:
Nice job! I have a few nitpicks regarding the presentation:

* [ ] The Non-Python packages link to the spec file. That's a useful link, but it's not the only useful one; it would be nice to link to the Fedora Package Database as well (e.g. [kajongg](https://src.fedoraproject.org/cgit/rpms/kajongg.git/tree/kajongg.spec)).
Maybe you can use a table, like the one for Mispackaged packages on the main page, and add the two links in columns?

* [ ] The small "progress bar" on the main page (below Naming Policy in the sidebar) is hard to read: the naming policy colors are too dark for contrast with black text. 
It should be OK to remove the text from all of these progress bars; it's not really useful in the other ones.

* [ ] In the same sidebar, the fields with explanations need the little color boxes to visually link them to the progressbar.

* [ ] I'm not sold on using the Naming policy colors for the vertical bars of color before the package names. Could you try to use the regular status colors, even here? That color tends to follow the package across portingdb.
  * [ ] If you decide to keep the naming policy colors: the vertical bar should always be on the same line as the package name. (For regular status colors the macro should ensure this.)

One thing I wanted to ask: what happens if a package has both a **Misnamed Subpackage** and **Misnamed Requires**?

irushchyshyn commented:
Thanks a lot for the review @encukou !

>     The Non-Python packages link to the spec file. That's a useful link, but it's not the only useful one; it would be nice to link to the Fedora Package Database as well (e.g. kajongg).
>     Maybe you can use a table, like the one for Mispackaged packages on the main page, and add the two links in columns?

That would be useful, I agree. I will try adding the table and will see how it looks.

>     The small "progress bar" on the main page (below Naming Policy in the sidebar) is hard to read: the naming policy colors are too dark for contrast with black text.
>     It should be OK to remove the text from all of these progress bars; it's not really useful in the other ones.

Agreed, will remove the text.

>     In the same sidebar, the fields with explanations need the little color boxes to visually link them to the progressbar.

I will fix it.

>     I'm not sold on using the Naming policy colors for the vertical bars of color before the package names. Could you try to use the regular status colors, even here? That color tends to follow the package across portingdb.

I thought it would be a bit confusing, but I will try to keep the original colors there and we will see.

> One thing I wanted to ask: what happens if a package has both a Misnamed Subpackage and Misnamed Requires?

Good question, I thought of it for a while too. In this case, the package will go into `Misnamed Subpackage` category, as it seems more important to unblock the packages which depend on it. Only after the name is fixed, it will got into `Misnamed Requires` category, if it still has the issue. Do you have another idea about it?

encukou commented:
> > One thing I wanted to ask: what happens if a package has both a Misnamed Subpackage and Misnamed Requires?
> 
> Good question, I thought of it for a while too. In this case, the package will go into `Misnamed Subpackage` category, as it seems more important to unblock the packages which depend on it. Only after the name is fixed, it will got into `Misnamed Requires` category, if it still has the issue. Do you have another idea about it?

That implies a workflow I'd rather avoid: us opening a bug for *Misnamed Subpackage*, waiting for the maintainer to fix, and then opening a bug for the *Misnamed Requires* – even though they could both be fixed in one update.

Could you just note this “overriding” on the Naming Policy Violations page? The raw data is there, we need to keep this in mind for a mass bug filing script, but I don't think you need to spend time on visualising it perfectly.

irushchyshyn commented:
Of course, I will add a note.

irushchyshyn commented:
Review updates are available in 4a80cc1:

- [x] Removed text from the small progress bars;
- [x] Added little color boxes for Naming policy sidebar legends to link them to the progress bar:
![image](https://cloud.githubusercontent.com/assets/5603118/25905151/8076a8aa-35a1-11e7-8a9d-253f605738b2.png)
- [x] Used regular status colors for vertical bars before the package names. Shown in the screen below, does it look better?
- [x] Added a note on "overriding" the Misnamed Requires status for packages  with misnamed subpackages:
![image](https://cloud.githubusercontent.com/assets/5603118/25905235/bdaac954-35a1-11e7-8e8b-fb5b7d4c5445.png)
- [x] For Non-Python Packages, also added a link to Fedora Package DB, and displayed as a table:
![image](https://cloud.githubusercontent.com/assets/5603118/25905354/1d0de368-35a2-11e7-8ac6-ca1d0684309d.png)

hroncok commented:
> Used regular status colors for vertical bars before the package names. Shown in the screen below, does it look better?

Not for my OCD :D

irushchyshyn commented:
I tend to agree. So I decided to keep the naming policy colors, but fixed the vertical bar position. Amended to  b0c1d75

>  If you decide to keep the naming policy colors: the vertical bar should always be on the same line as the package name. (For regular status colors the macro should ensure this.)

encukou commented:
I've added a fix for one more nitpick (directly to the PR, to save a review round-trip).
If you agree, let's merge this.

irushchyshyn commented:
Thanks @encukou , it looks good.
